src/main/java/ch/uzh/ifi/hase/soprafs24/service/GameService.java

```diff
@@ -341,16 +340,19 @@ public void onGameCalculationStatusChange(InactiveRequestEvent event) {
     }
 
     private void executeInactivePlayerLogic(Integer gameId) {
-        User inactivePlayer = UserContextHolder.getCurrentUser();
         Game currentGame = inMemoryGameRepository.findById(gameId);
 
-
         int historySize = currentGame.getHistory().size();
 
         // We need to ensure that a correct selection at the very end of a turn does not in a wrongful change of Turn
-        if (historySize > 1 &&
+        if (!(historySize > 1 &&
                 (currentGame.getHistory().get(historySize - 1).getPicks().isEmpty()
-                        && currentGame.getHistory().get(historySize - 2).getUserId() == currentGame.getActivePlayer())) {return;}
+                        && currentGame.getHistory().get(historySize - 2).getUserId() == currentGame.getActivePlayer()))) {updateOnInactivity(gameId);}
```

Comment:
I just realized that if the user is ongoing for streak third time, (henry - henry - henry's turn) then does this if condition work in this case??

If this is not the case, quick thought is that from the frontend, send mid time REST request to update boolean value (nowPlaying) and check whether this request was sent to identify this to activaet updateOnInactivity?
in normal turn passing, we can update the boolean value back at the initiateNewTurn.

Reply:
that should have no impact. It is only necessary to know that the current turn (which is empty) and the last are from the same person